java/com/google/flatbuffers/Comparators.java

```diff
@@ -0,0 +1,45 @@
+package com.google.flatbuffers;
+
+/**
+ * Utilities function to signed and unsigned compare values in their natural order
+ */
+public class Comparators {
+    private static final int UBYTE_MASK = 0xff;
+    private static final int USHORT_MASK = 0xffff;
+    private static final int UINT_SIGN_BIT = 0x80000000;
+    private static final long ULONG_SIGN_BIT = 0x8000000000000000L;
+
+    public static int compare(long a, long b) {
+        return a > b ? 1 : a < b ? -1 : 0;
+    }
+
+    public static int compare(int a, int b) {
+        return a > b ? 1 : a < b ? -1 : 0;
```

Comment:
I managed to modify the code to always use comparison directly. This class was renamed unsigned and now only perform the conversion required to compared unsigned number using signed operators.